### PR TITLE
Add new `driftfile`, `hwtimestamps`, `rtcsync`, and `dumpdir` parameters

### PR DESCRIPTION
**Pull request by chrekh**
The params added with this PR are:
driftfile, hwtimestamps, rtcsync, and dumpdir. Each with it's separate commit.

### CODE REVIEW

**Comment by chrekh** [comment]
Rebased to master/HEAD because the need to add defaultvalue for $dumpdir for Gentoo in params.pp

File: REFERENCE.md [hunk 1]
```diff
@@ -475,3 +483,35 @@ See [clientloglimit](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#clien
 
 Default value: `undef`
 
+##### `rtcsync`
+
+Data type: Boolean
+
+Periodically sync syste time to RTC
```
Comment (alexjfisher) [inline REFERENCE.md:490]: typo in `system`

File: REFERENCE.md [hunk 2]
```diff
@@ -475,3 +483,35 @@ See [clientloglimit](https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#clien
...
+Keep RTC in UTC instead of local time.
+
+Default value: $chrony::params::rtconutc
+
+##### `hwtimestamps`
+
+Data type: `Variant[Hash,Array[String]]`
+
+This selects interfaces to enable hardware timestamps on. It can be an array of interfaces
+or a hash of interfaces to ther respective options.
```
Comment (alexjfisher) [inline REFERENCE.md:507]: typo here too.  The code looks correct though.  I guess REFERENCE.MD just needs regenerating?

File: templates/chrony.conf.epp [hunk 3]
```diff
@@ -120,3 +128,12 @@ smoothtime <%= $chrony::smoothtime %>
 # https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html#rtconutc
 rtconutc
 <% } -%>
+<% if ! $chrony::hwtimestamps.empty { -%>
```
Comment (alexjfisher) [inline templates/chrony.conf.epp:131]: Would `unless` be more readable than `if !`?

File: REFERENCE.md (hunk 1 above)
Comment (chrekh) [reply REFERENCE.md:490]: Yes, good catch.  Do you prefer that I add a small typo-fix commit, or a rebase -i and a force-push?

**Comment by alexjfisher** [comment]
Would you be able to add any extra tests to cover the new options?  In particular, it'd be good to test that `hwtimestamps` works correctly with both Hash and Array values.

File: templates/chrony.conf.epp (hunk 3 above)
Comment (chrekh) [reply templates/chrony.conf.epp:131]: Interesting that you should mention that. I myself prefer unless before if ! in any language that supports it. But, I know people that hates unless. :)

**Comment by chrekh** [comment]
> Would you be able to add any extra tests to cover the new options? In particular, it'd be good to test that `hwtimestamps` works correctly with both Hash and Array values.

Yes, I can do that.

**Comment by chrekh** [comment]
I have rebased to master, fixed the typos, and added tests in the commits for each param

File: templates/chrony.conf.epp (hunk 3 above)
Comment (chrekh) [reply templates/chrony.conf.epp:131]: There are more 'if !' in the template, I can add a separate PR later that changes them all to unless, and se if someone else objects.

**Pull request merged**